### PR TITLE
404 on label lookup failure

### LINKED ISSUE #942: 404s not handled correctly on url with bad linode label
Try going to the linode page using a non-existing linode id/label

http://localhost:3000/linodes/1

The page should 404 or redirect to the /linodes page.

### PR DESCRIPTION
**Pull request by eatonphil**
Closes #942 (and solves the same problem for all other sections that do a label lookup)

### CODE REVIEW

File: src/api/util.js [hunk 1]
```diff
@@ -10,10 +19,16 @@ export function getObjectByLabelLazily(pluralName, label, labelName = 'label') {
...
     }
 
-    return (await dispatch(api[pluralName].all([], undefined, {
+    const response = (await dispatch(api[pluralName].all([], undefined, {
       headers: {
         'X-Filter': JSON.stringify({ [labelName]: label }),
       },
     })))[pluralName][0];
+
+    if (!response) {
```
Comment (mparke) [inline src/api/util.js]: this assumes a 404?

Comment (eatonphil) [inline src/api/util.js]: can't think of a way it wouldn't be.

Comment (mparke) [inline src/api/util.js]: how do we handle 500?

Comment (eatonphil) [inline src/api/util.js]: 500s are already handled by the fetch library. In the case of search by label, the results being empty means that it was essentially a 404.

Comment (mparke) [inline src/api/util.js]: is 500 handling at a different level, before this returns?

Comment (eatonphil) [inline src/api/util.js]: Yes

Comment (mparke) [inline src/api/util.js]: okay, I see that `status >= 400` is rejected, but how is it handled in the UI?

also, I feel like this implicit empty object return means 404 is kind of ambiguous, is there a better way we can represent this?

Comment (eatonphil) [inline src/api/util.js]: > okay, I see that status >= 400 is rejected, but how is it handled in the UI?

Look for `errors` in `src/layouts/Layout` or `src/index.js`.

File: src/api/util.js [hunk 2]
```diff
@@ -10,10 +19,16 @@ export function getObjectByLabelLazily(pluralName, label, labelName = 'label') {
...
 
-    return (await dispatch(api[pluralName].all([], undefined, {
+    const response = (await dispatch(api[pluralName].all([], undefined, {
       headers: {
         'X-Filter': JSON.stringify({ [labelName]: label }),
       },
-    })))[pluralName][0];
+    })));
+
+    if (!response.total_results) {
```
Comment (mparke) [inline src/api/util.js]: total_results could be 0 and this be a 200, no?

**Pull request merged**